src/file_handle/native.rs

```diff
@@ -57,6 +57,55 @@ impl Future for Reader {
     }
 }
 
+struct WriterState {
+    bytes: Option<Box<[u8]>>,
+    waker: Option<Waker>,
+}
+
+struct Writer {
+    state: Arc<Mutex<WriterState>>,
+}
+
+impl Writer {
+    fn new(path: &Path, bytes: Box<[u8]>) -> Self {
+        let state = Arc::new(Mutex::new(WriterState {
+            bytes: Some(bytes),
+            waker: None,
+        }));
+
+        {
+            let path = path.to_owned();
+            let state = state.clone();
+            std::thread::Builder::new()
+                .name("rfd_file_write".into())
+                .spawn(move || {
+                    let mut state = state.lock().unwrap();
+                    let _res = std::fs::write(path, state.bytes.take().unwrap());
```

Comment:
Wouldn't holding the lock here during the whole write operation cause calls to poll to block?

Reply:
Fixed in [6284197](https://github.com/PolyMeilex/rfd/pull/134/commits/62841970485454a8c2da21e0d36ba2f526917d1b)